server/app/services/applicant/ApplicantService.java

```diff
@@ -465,6 +469,175 @@ public ImmutableList<Application> getApplications(TimeFilter submitTimeFilter) {
+                                      return a.getSubmitTime() != null
+                                          ? a.getSubmitTime()
+                                          : Instant.ofEpochMilli(0);
+                                    })
+                                .thenComparing(Application::getCreateTime)))));
+
+    ImmutableList.Builder<ApplicantProgramData> inProgressPrograms = ImmutableList.builder();
+    ImmutableList.Builder<ApplicantProgramData> submittedPrograms = ImmutableList.builder();
+    ImmutableList.Builder<ApplicantProgramData> unappliedPrograms = ImmutableList.builder();
+
+    Set<String> programNamesWithApplications = Sets.newHashSet();
+    mostRecentApplicationsByProgram.forEach(
+        (programName, appByStage) -> {
+          Optional<Application> maybeDraftApp =
+              appByStage.getOrDefault(LifecycleStage.DRAFT, Optional.empty());
+          Optional<Application> maybeSubmittedApp =
+              appByStage.getOrDefault(LifecycleStage.ACTIVE, Optional.empty());
+          Optional<Instant> latestSubmittedApplicationTime =
+              maybeSubmittedApp.map(Application::getSubmitTime);
+          if (maybeDraftApp.isPresent()) {
+            inProgressPrograms.add(
+                ApplicantProgramData.create(
+                    maybeDraftApp.get().getProgram().getProgramDefinition(),
+                    latestSubmittedApplicationTime));
+            programNamesWithApplications.add(programName);
+          } else if (maybeSubmittedApp.isPresent() && activeProgramNames.containsKey(programName)) {
+            ProgramDefinition programDef = activeProgramNames.get(programName);
+            submittedPrograms.add(
+                ApplicantProgramData.create(programDef, latestSubmittedApplicationTime));
+            programNamesWithApplications.add(programName);
+          }
+        });
+
+    Set<String> unappliedActivePrograms =
+        Sets.difference(activeProgramNames.keySet(), programNamesWithApplications);
+    unappliedActivePrograms.forEach(
+        programName -> {
+          unappliedPrograms.add(
+              ApplicantProgramData.create(activeProgramNames.get(programName), Optional.empty()));
+        });
+
+    // Ensure each list is ordered by database ID for consistent ordering.
+    return RelevantPrograms.builder()
+        .setInProgress(sortByProgramId(inProgressPrograms.build()))
+        .setSubmitted(sortByProgramId(submittedPrograms.build()))
+        .setUnapplied(sortByProgramId(unappliedPrograms.build()))
+        .build();
+  }
+
+  private ImmutableList<ApplicantProgramData> sortByProgramId(
+      ImmutableList<ApplicantProgramData> programs) {
+    return programs.stream()
+        .sorted(Comparator.comparing(p -> p.program().id()))
+        .collect(ImmutableList.toImmutableList());
+  }
+
+  private void checkForDuplicateDrafts(ImmutableSet<Application> applications) {
+    // Add logging to better understand a bug where some applicants were seeing
```

Comment:
nit: Use active language.

The logging is to better understand....

Also perhaps this should just be the method javadoc since it doesn't check/throw like our other check methods doo (checkNotNull) etc.  maybe rename, logDuplicateDrafts etc

Reply:
Done. Used more active language, moved to javadoc, and renamed to `logDuplicateDrafts`.